### PR TITLE
feat: ✨ add types for chartjs-color-string

### PR DESCRIPTION
**Pull request by Ankan002**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/chartjs-color-string/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Type definitions for chartjs-color-string 0.6
+// Project: https://github.com/chartjs/chartjs-color-string#readme
+// Definitions by: Ankan Bhattacharya <https://github.com/Ankan002>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare const colorString: colorString.ColorString;
+
+declare namespace colorString {
```
Comment (jakebailey) [inline types/chartjs-color-string/index.d.ts:8]: There's no reason to use the namespace pattern here; the upstream package just has an export object, which can be typed on DT plainly as:

```ts
export function getRgba(...): ...;
export function getHsla(...): ...;
// ...
```

Comment (jakebailey) [reply types/chartjs-color-string/index.d.ts:8]: For reference, you only ever need the namespace pattern if the exported object needs to be something that's "merged", usually a function that they've added properties to.

Comment (Ankan002) [reply types/chartjs-color-string/index.d.ts:8]: Okk but would it support commonjs imports as it has been exported in the format module.exports

Comment (jakebailey) [reply types/chartjs-color-string/index.d.ts:8]: Yes. The only reason to use `export =` is when the exported thing _isn't_ just a plain object.

Comment (Ankan002) [reply types/chartjs-color-string/index.d.ts:8]: Okk then we can simply export the object as default export and the interface as a named export

```typescript
export interface ColorString {
        getRgba(colorString?: string): number[] | undefined;
        getHsla(colorString?: string): number[] | undefined;
        getHwb(colorString?: string): number[] | undefined;
        getRgb(colorString?: string): number[] | undefined;
        getHsl(colorString?: string): number[] | undefined;
        getAlpha(colorString?: string): number | undefined;

        hexString(rgba: number[], a?: number): string;
        rgbString(rgba: number[], alpha?: number): string;
        rgbaString(rgba: number[], alpha?: number): string;
        percentString(rgba: number[], alpha?: number): string;
        percentaString(rgba: number[], alpha?: number): string;
        hslString(hsla: number[], alpha?: number): string;
        hslaString(hsla: number[], alpha?: number): string;
        hwbString(hwb: number[], alpha?: number): string;
        keyword(rgb: number[]): string;
    }
    
export default colorString: ColorString;
```

Comment (Ankan002) [reply types/chartjs-color-string/index.d.ts:8]: Or should I follow this format

```typescript
export function getRgba(...): ...;
export function getHsla(...): ...;
```

@jakebailey

Comment (jakebailey) [reply types/chartjs-color-string/index.d.ts:8]: The latter, i.e.:

```ts
export function getRgba(colorString?: string): number[] | undefined;
export function getHsla(colorString?: string): number[] | undefined;
export function getHwb(colorString?: string): number[] | undefined;
export function getRgb(colorString?: string): number[] | undefined;
// ...
```

Comment (Ankan002) [reply types/chartjs-color-string/index.d.ts:8]: Got it

Comment (Ankan002) [reply types/chartjs-color-string/index.d.ts]: @jakebailey just implemented the required changes

**Review by jakebailey — APPROVED** [review]

**Comment by Ankan002** [comment]
Ready to merge

**Pull request merged**